larpandoracontent/LArObjects/LArMCParticle.h

```diff
@@ -27,7 +27,9 @@ namespace lar_content
 class LArMCParticleParameters : public object_creation::MCParticle::Parameters
 {
 public:
-    pandora::InputInt   m_nuanceCode;               ///< The nuance code
+    pandora::InputInt                          m_nuanceCode;        ///< The nuance code
+    std::vector<pandora::InputCartesianVector> m_mcStepPositions;   ///< The positions of the geant4 steps
```

Comment:
I added a +1 to @AndyChappell's related comments, but expand within this review. Basically, we prefer pandora::InputTypes (e.g. pandora::InputCartesianVector) in the provision of input parameters (although there is an overhead) because this automatically guards against uninitialised values and INFs and NaNs. Within the target, initialised object, though, prefer the actual raw types (e.g. pandora::CartesianVector).